kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/util/CachedTokenObject.java

```diff
@@ -0,0 +1,18 @@
+package io.mosip.kernel.auth.defaultadapter.util;
+
+import lombok.Getter;
+
+public class CachedTokenObject<T> {
+
+	@Getter
+	private T token;
+
+	public void setToken(T token) {
```

Comment:
Should we make this method synchronized since there is a possibility of both task scheduler and SelfTokenRestInterceptor calling at the same time ?

Reply:
When the task scheduler is running there is no possibility of SelfTokenRestInterceptor to call this set token.

Reply:
what if there is some issue and task scheduler stopped running ?